base/docs/basedocs.jl

```diff
@@ -386,56 +381,84 @@ desired can be used.
 kw"if", kw"elseif", kw"else"
 
 """
-`for` loops repeatedly evaluate the body of the loop by iterating over a sequence of
-values. For example:
+    for
 
-    for i in [1,4,0]
-        println(i)
-    end
+`for` loops repeatedly evaluate the body of the loop by
+iterating over a sequence of values.
+
+# Examples
+```jldoctest
+julia> for i in [1, 4, 0]
+           println(i)
+       end
+1
+4
+0
+```
 """
 kw"for"
 
 """
+    while
+
 `while` loops repeatedly evaluate a conditional expression, and continues evaluating the
 body of the while loop so long as the expression remains `true`. If the condition
 expression is false when the while loop is first reached, the body is never evaluated.
-For example:
 
-    while i <= 5
-        println(i)
-        i += 1
-    end
+# Examples
+```jldoctest
+julia> i = 1
+1
+
+julia> while i < 5
+           println(i)
+           i += 1
+       end
+1
+2
+3
+4
+```
 """
 kw"while"
 
 """
-`end` marks the conclusion of a block of expressions. In the example below, `end` marks
-the conclusion of a `function`.
-
-    function foo()
-        println("hello, world")
     end
 
-`end` marks the conclusion of all kinds of expression blocks: `module`, `type`, `begin`,
-`let`, `for`, etc.
+`end` marks the conclusion of a block of expressions, for example
+`module`, `struct`, `mutable struct`, `begin`, `let`, `for` etc.
+`end` may also be used when indexing into an array to represent
+the last index of a dimension.
 
-In addition, `end` may be used when indexing into an array to represent the last index
-of each dimension:
-
-    x[1:end, 2:end-1]
+# Examples
+```jldoctest
+julia> A = [1 2; 3 4]
+2×2 Array{Int64,2}:
+ 1  2
+ 3  4
+
+julia> A[end, :]
+2-element Array{Int64,1}:
+ 3
+ 4
+```
 """
 kw"end"
 
 """
+    try/catch
+
 A `try/catch` statement allows for `Exception`s to be tested for. For example, a
```

Comment:
I'd format this as `` `try`/`catch` ``, since they're separate keywords. Also if you're using a slash here, I'd do the same for `if`/`elseif`/`else`.